File: code-coaches/medium/the-spy-life/the_spy_life.py
def decode_message(message: str) -> str:
    message_without_symbols = "".join(
        [char for char in message if char.isalpha() or char.isspace()]
    )
    flipped_message_without_symbols = message_without_symbols[::-1]
    return flipped_message_without_symbols


def main():
    encoded_message = input()
    print(decode_message(encoded_message))


if __name__ == "__main__":
    main()

File: code-coaches/medium/the-spy-life/the_spy_life.rb
def decode_message(message)
    message_without_symbols = message.chars.select { |char| char.match?(/\s|[[:alpha:]]/) }.join
    flipped_message_without_symbols = message_without_symbols.reverse
    return flipped_message_without_symbols
end


def main()
    encoded_message = gets.chomp
    puts decode_message(encoded_message)
end


if __FILE__ == $0
    main()
end
